scripts/ice-road-pipeline.py

```diff
@@ -124,15 +164,170 @@
+
+    # Write reproject-match SnowDepth and SnowOn to disk
     snowdepth.rio.to_raster(snow_depth_path)
+    snowon_matched.rio.to_raster(snow_tif)
 
     # difference two rasters to find canopy height
     ref_dem_path = join(results_dir, 'dem.tif')
     canopy_fp = join(ice_dir, f'{basename(in_dir)}-canopyheight.tif')
     canopy = rio.open_rasterio(canopy_tif, masked=True) 
     matched = canopy.rio.reproject_match(snowoff)
     canopyheight = matched - snowoff
-    # mask snow depth from vegetation
-    canopyheight = canopyheight.where((canopyheight > snowon + 0.1) | (snowon.isnull()))
+
+    # mask snow depth from vegetation (null=0)
+    canopyheight = canopyheight.where((canopyheight > snowdepth + 0.1) | (snowdepth.isnull()), other=0)
+    canopyheight.rio.to_raster(canopy_fp)
+
+    # Command for testing...
+    # python ice-road-pipeline.py /Users/brent/Code/ice-road-copters/data/feb9/mcs/ -e /Users/brent/Code/LIDAR/data/QSI_snowfree.tif -a /Users/brent/Code/ice-road-copters/ASP/ -s /Users/brent/Code/ice-road-copters/transform_area/hwy_21/hwy21_utm_edit_v3.shp -r /Users/brent/Code/ice-road-copters/transform_area/Eagle/hill_rd_55_intersection.shp -i /Users/brent/Documents/MCS/LIDAR/IMU/20230209_MCS_precise_scanner_bw.txt -d True -c /Users/brent/Code/LIDAR/data/fl_230210_002840/20230209_extraBytes-230210_002840_Scanner_1.las -k 0.1956
+    if shp_fp_rfl:
+
+        # Make sure R is installed
+        proc = Popen(["which", "R"],stdout=PIPE,stderr=PIPE)
+        exit_code = proc.wait()
+        if exit_code != 0:
+            raise Exception("Please install R on this system and packages: raster, lidR, rlas, dplyr, readr, terra,sf, data.table.")
+
+        # Check whether dem directory exists, if not, make one
+        ssa_dir = f'{results_dir}/ssa-calc'
+        if not os.path.exists(ssa_dir):
+            os.makedirs(ssa_dir)
+
+        # Displaying the parent directory of the script
+        scripts_dir = os.path.dirname(__file__)
+
+        # Load in the ASP translational-only matrix from pc-align tool
+        asp_matrix_fp = glob(f'{ice_dir}/results/pc-align-translation-only/temp-log*.txt')[0]
+        with open(asp_matrix_fp, 'r') as f:
+            lines = f.readlines()
+            for line in lines:
+                if "Translation vector (North-East-Down, meters): Vector3" in line: 
+                    n_e_d_shift = re.findall("[+-]?\d+\.\d+", line)
+                    n_e_d_shift = ','.join(n_e_d_shift)
+                    break
+
+        # get crs
+        ref_raster = rasterio.open(snow_tif)
+        crs = ref_raster.crs
+        pix_size, _ = ref_raster.res
+        crs = str(crs).split(":",1)[1]    
+        ras_meta = ref_raster.profile
+
+        # Compute slope and aspect from snow-on lidar
+        slope_fp = join(ice_dir, f'{basename(in_dir)}-snowon_slope.tif')
+        aspect_fp = join(ice_dir, f'{basename(in_dir)}-snowon_aspect.tif')
+        os.system(f'gdaldem slope -compute_edges {snow_tif} {slope_fp} -q')
```

Comment:
Another option to consider - https://xarray-spatial.org/reference/_autosummary/xrspatial.slope.slope.html
https://xarray-spatial.org/reference/_autosummary/xrspatial.aspect.aspect.html

Reply:
If you want to keep it all in xarray I really like xrspatial.

Reply:
Hmm yeah I agree with you here, especially since we don't need to save these rasters.

@ZachHoppinen There are two other instances of calling gdal in laz_align.py for reprojection for geoid conversion (which I also noticed assume EPSG 32611 with no user input). Also how can we programmatically tell the equivalence between same WGS and NAD83 projs... 

Now hear me out, what if we just broke that part entirely out of the pipeline and made it a separate utility script? That way the pipeline can just expect to see the DEM in WGS only.

Reply:
Actually maybe leaving the laz_align alone for right now is okay. We can circle back if need be and we want to make it more transferable to other sites.